docs/releases/4.2.md

```diff
@@ -245,3 +251,18 @@ If you are using the front-end cache invalidator module (`wagtail.contrib.fronte
 * For Azure Front Door: upgrade `azure-mgmt-frontdoor` to version 1 or above
 
 Support for older versions will be dropped in a future release.
+
+### Changes to `Workflow` and `Task` methods
+
+To accommodate workflows support for snippets, the `page` argument in {meth}`Workflow.start() <wagtail.models.Workflow.start>` has been renamed to `obj`.
+
+In addition, some methods on the base {class}`~wagtail.models.Task` model have been changed.
```

Comment:
I think this is the most important bit, as we expect customisations here, whereas we don't elsewhere in workflow, so I think worth explicitly mentioning that this is relevant for custom task models. Also, I think you'll need to update https://github.com/laymonage/wagtail/blob/snippets-workflow/docs/extending/custom_tasks.md to match, and perhaps link it here.